RxCocoa/Traits/Publisher.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  Publisher.swift
+//  RxCocoa
+//
+//  Created by Krunoslav Zaher on 3/28/15.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+import Dispatch
+#if !RX_NO_MODULE
+    import RxSwift
+#endif
+
+/// Publisher is a wrapper for `PublishSubject`.
+///
+/// Unlike `PublishSubject` it can't terminate with error, and when publisher is deallocated
+/// it will complete it's observable sequence (`asObservable`).
+public final class Publisher<Element> {
```

Comment:
Let's name this like @sergdort suggests `PublishRelay`.

Maybe it would also make sense to create something like:
```
enum Relay {
    class Publish<Element> {
       ...;
    }
}
``` 

Consistent with https://github.com/JakeWharton/RxRelay.

We will also add `BehaviorRelay` and deprecate `Variable`. (but that's separate work).